lib/core/logger.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <iostream>
+#include <optional>
+#include <string>
+
+#include "time.hpp"
+
+namespace hyped::core {
+
+enum class LogLevel { kNone = 0, kDebug, kInfo, kFatal };
+
+class ILogger {
+ public:
+  virtual void log(const LogLevel level, const char *format ...){};
```

Comment:
`=0`

also check the ROS interface but it may now be possible to put a `const` at the end here, too.